src/message/mailbox/parsers/rfc2822.rs

```diff
@@ -0,0 +1,146 @@
+fn text() -> impl Parser<char, char, Error = Simple<char>> {
+    one_of(TEXT.as_slice())
+}
+
+// 3.2.2. Quoted characters
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.2
+
+// quoted-pair     =       ("\" text) / obs-qp
+fn quoted_pair() -> impl Parser<char, char, Error = Simple<char>> {
+    just('\\').ignore_then(text())
+}
+
+// 3.2.4. Atom
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.4
+
+static ATEXT: Lazy<Vec<char>> = Lazy::new(|| {
+    let mut range: Vec<char> = vec![];
+    range.extend('0'..='9');
+    range.extend('a'..='z');
+    range.extend('A'..='Z');
+    range.extend("!#$%&\'*+-/=?^_`{|}~".chars());
+    range
+});
+
+// atext           =       ALPHA / DIGIT / ; Any character except controls,
+//                         "!" / "#" /     ;  SP, and specials.
+//                         "$" / "%" /     ;  Used for atoms
+//                         "&" / "'" /
+//                         "*" / "+" /
+//                         "-" / "/" /
+//                         "=" / "?" /
+//                         "^" / "_" /
+//                         "`" / "{" /
+//                         "|" / "}" /
+//                         "~"
+pub(super) fn atext() -> impl Parser<char, char, Error = Simple<char>> {
+    one_of(ATEXT.as_slice())
+}
+
+// 3.2.5. Quoted strings
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.5
+
+static QTEXT: Lazy<Vec<char>> = Lazy::new(|| {
+    let mut range: Vec<char> = vec![33 as char];
+    range.extend((35 as char)..=(91 as char));
+    range.extend((93 as char)..=(126 as char));
+    range
+});
+
+// qtext           =       NO-WS-CTL /     ; Non white space controls
+//
+//                         %d33 /          ; The rest of the US-ASCII
+//                         %d35-91 /       ;  characters not including "\"
+//                         %d93-126        ;  or the quote character
+fn qtext() -> impl Parser<char, char, Error = Simple<char>> {
+    one_of(QTEXT.as_slice())
+}
+
+// qcontent        =       qtext / quoted-pair
+pub(super) fn qcontent() -> impl Parser<char, char, Error = Simple<char>> {
+    choice((qtext(), quoted_pair()))
+}
+
+// 3.4. Address Specification
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.4
+
+// mailbox         =       name-addr / addr-spec
+pub(crate) fn mailbox(
+) -> impl Parser<char, (Option<String>, (String, String)), Error = Simple<char>> {
+    choice((addr_spec().map(|addr| (None, addr)), name_addr()))
+}
+
+// name-addr       =       [display-name] angle-addr
+fn name_addr() -> impl Parser<char, (Option<String>, (String, String)), Error = Simple<char>> {
+    rfc2234::dquote()
+        .or_not()
+        .ignore_then(
+            // NOTE: take everything available between potential
+            // quotes in order to make the parsing of the display-name
+            // the most flexible possible
+            take_until(rfc2234::dquote().or_not().ignore_then(angle_addr())).map(
```

Comment:
Do we actually need to be this conservative? I think I would prefer to stick to the RFC. If this is because of `parse_multi_with_name_containing_comma` I think that test is just wrong. The "display  name' in that test should be quoted.

Reply:
I just refactored this part (to stick more to the RFC), and some tests fail (surprisingly not `parse_multi_with_name_containing_comma`). One of them is `parse_address_with_name`:

```rust
#[test]
fn parse_address_with_name() {
    assert_eq!(
        "K. <kayo@example.com>".parse(),
        Ok(Mailbox::new(
            Some("K.".into()),
            "kayo@example.com".parse().unwrap()
        ))
    );
}
```

The dot `.` is not a valid char in an `atom` (nor in a `uAtom`), so being more close to the RFC will lead to some "breaking" changes (some cases that use to pass with the previous parser will not with this new stricter one).

Reply:
:heavy_check_mark: